juniper/src/schema/schema.rs

```diff
@@ -282,11 +282,11 @@ impl<'a, S: ScalarValue + 'a> TypeType<'a, S> {
             TypeType::Concrete(&MetaType::Interface(InterfaceMeta {
                 name: ref iface_name,
                 ..
-            })) => Some(
-                context
-                    .concrete_type_list()
-                    .iter()
-                    .filter_map(|&ct| {
+            })) => {
+                let mut type_names: Vec<_> = context
+                    .types
+                    .values()
```

Comment:
This is an optimization. `concrete_type_list()` collects into a vector first, which is wasteful here.